### ISSUE TITLE
Public and Private Keys with Laravel Passport

### ISSUE THREAD

**Issue by medina325** [ronvanderheijden/openid-connect#8]
In this [section](https://github.com/ronvanderheijden/openid-connect#keys) you describe how to generate the keys for later encryption of the ID Token. Can I simply use the ones generated by the laravel artisan passport:keys command? If so, where do I store them for the openid provider to pick it up? Right now they don't appear in the token being generated.
![image](https://user-images.githubusercontent.com/56964034/135292299-c3bc1f95-363c-4e32-b21f-aedc31aaf368.png)

But if I add them , the signature becomes valid.

![image](https://user-images.githubusercontent.com/56964034/135292439-d7a930cf-ac3f-4c94-b069-e2a15b53bad6.png)

**Comment by medina325** [comment]
By the way, only the public key needs to be added in order for the signature to be verified.

I'm following the right steps, I verified the private key is being fetched correctly from /storage/oauth-private.key when the authorization server is created, but I don't know what's happening to the public key.

Would you please take a look into why the public key does not get attached to the id token? Thanks!

**Comment by andrewminion-luminfire** [comment]
@medina325 did you ever figure this out? I’m running into the same invalid signature.

I was using jwt.io to verify my JWT; it doesn’t allow me to specify a public and private key: just a secret. But whenever I added a secret, it would resign the JWT and change the signature.

The only way I was able to produce a verifiable signature was to change `makeAuthorizationServer()` in `src/Laravel/PassportServiceProvider.php` to use the oauth client secret rather than the private key file:

```diff
+ $validated = request()->validate([
+     'client_id' => ['required', 'string', 'max:36', 'uuid'],
+ ]);
+ $client = \Laravel\Passport\Client::find($validated['client_id']);
+
$responseType = new IdTokenResponse(
    app(config('openid.repositories.identity')),
    new ClaimExtractor(...$claimSets),
    Configuration::forSymmetricSigner(
        app(config('openid.signer')),
-        InMemory::file($cryptKey->getKeyPath()),
+        InMemory::plainText($client->secret),
    ),
);
```

It seems that the consumer expects the JWT to be signed by the (shared) client secret, not the private key file.

**Comment by andrewminion-luminfire** [comment]
@ronvanderheijden what do you think of adding a config option to specify whether the private key or the client secret should be used as the signing key?

**Comment by medina325** [comment]
@andrewminion-luminfire sorry for taking so long.
I couldn't solve the problem, and due to the tight schedule I had to give up on OpenID Connect and tried other protocols and tools.
I ended up using [this](https://github.com/codegreencreative/laravel-samlidp) implementation of the SAML SSO protocol, where my application acted as the "identity provider".

**Comment by aamiranwar001** [comment]
> @ronvanderheijden what do you think of adding a config option to specify whether the private key or the client secret should be used as the signing key?

Please change the signer option value from the OpenID config file.

'signer' => \Lcobucci\JWT\Signer\Rsa\Sha256::class

**Comment by andrewminion-luminfire** [comment]
@aamiranwar001 I think you’re talking about the signing algorithm, correct? I’m talking about the contents of the key itself.

**Comment by aamiranwar001** [comment]
@ronvanderheijden Yes, I'm talking about the signing algorithm. \Lcobucci\JWT\Signer\Rsa\Sha256::class algorithm has options for public and private keys. Besides this, you're correct. We need a config option for client secret.

**Comment by ronvanderheijden** [comment]
I'm very sorry, but I'm currently too busy with some other stuff.
I also haven't touched this project for some time now.

I have planned some work on this project in the future.
You can always submit a pull-request, I will make time to review this. 

But for me to add features or fix bugs will have to wait.

**Comment by jeremy379** [comment]
Hello

I made a fork to make this package Laravel compliant (and Laravel first): https://github.com/jeremy379/laravel-openid-connect

It has some change/breaking change with this one, and it's made to work only with Laravel. 

But if it may help you our help you debugging, you're free to use it.
